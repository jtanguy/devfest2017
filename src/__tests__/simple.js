const simple = require('../simple.js');
const MacaroonsVerifier = require('macaroons.js').MacaroonsVerifier;
const TimestampCaveatVerifier = require('macaroons.js').verifier.TimestampCaveatVerifier;

describe('macaroon:creation', () => {
  test('m has the correct signature"', () => {
    expect(simple.m.signature).toMatchSnapshot();
  });

  test('m2 has the correct signature"', () => {
    expect(simple.m2.signature).toMatchSnapshot();
  });

  test('m3 has the correct signature"', () => {
    expect(simple.m3.signature).toMatchSnapshot();
  });
});

describe('macaroon:verification', () => {
  test('m is valid with the correct secret', () => {
    let verifier = new MacaroonsVerifier(simple.m);
    expect(verifier.isValid(simple.secret)).toBe(true);
  });
  test('m is not valid with the wrong secret', () => {
    let verifier = new MacaroonsVerifier(simple.m);
    expect(verifier.isValid('something else')).toBe(false);
  });
  test('m2 is not valid without verifying its caveats', () => {
    let verifier = new MacaroonsVerifier(simple.m2);
    expect(verifier.isValid(simple.secret)).toBe(false);
  });
  test('m2 is valid when verifying its caveats', () => {
    let verifier = new MacaroonsVerifier(simple.m2).satisfyExact(simple.c);
    expect(verifier.isValid(simple.secret)).toBe(true);
  });
  test('m2 is still valid when verifying more than its caveats', () => {
    let verifier = new MacaroonsVerifier(simple.m2)
      .satisfyExact(simple.c)
      .satisfyGeneral(TimestampCaveatVerifier);
    expect(verifier.isValid(simple.secret)).toBe(true);
  });
  test('m3 is valid when verifying its caveats', () => {
    let verifier = new MacaroonsVerifier(simple.m3)
      .satisfyExact(simple.c)
      .satisfyGeneral(TimestampCaveatVerifier);
    expect(verifier.isValid(simple.secret)).toBe(true);
  });
  test('m4 is not valid when verifying its caveats', () => {
    let verifier = new MacaroonsVerifier(simple.m4)
      .satisfyExact(simple.c)
      .satisfyGeneral(TimestampCaveatVerifier);
    expect(verifier.isValid(simple.secret)).toBe(false);
  });

});
