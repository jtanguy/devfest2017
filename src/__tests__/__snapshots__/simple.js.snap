// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macaroon:creation m has the correct signature" 1`] = `"7339954339bb20b9ceeb0cd4fff91f66ddbe6cc824efa49f165cbd950bc1dc1f"`;

exports[`macaroon:creation m2 has the correct signature" 1`] = `"c247c5613203141be46a14008678a2aaec41d83c9cf69d7b10b5c4e533047abb"`;

exports[`macaroon:creation m3 has the correct signature" 1`] = `"4bd18fd10c0193ae25fc73ce3f89b46a64ffb6508025ba3faeb5acf2656a99f6"`;
